lib/index.js

```diff
@@ -10,23 +10,41 @@ const Joi = require('joi');
 const Schema = require('./schema');
 const Package = require('../package.json');
 
-// const SchwiftyMigration = require('schwifty-migration');
-
 const internals = {};
 
 exports.register = function (server, options, next) {
 
+    Joi.assert(options, Schema.plugin, 'Bad plugin options passed to schwifty.');
+
     const rootState = internals.state(server.root);
 
-    internals.state(server).dir = options.dir;
     const knexFile = options.knexFile;
 
-    internals.state(server).knex = Knex(knexFile[process.env.NODE_ENV]);
+    /*
+        Move the models into the knexFile object
+        Since a models path isn't part of it's schema on knexjs.org
+    */
+
+    if (options.models) {
+        knexFile.models = options.models;
+    }
+
+    let knex;
+
+    try {
+        knex = Knex(knexFile[process.env.NODE_ENV]);
+    }
+    catch (err) {
```

Comment:
Any error involved with connecting to the db, or bad knexfile config options passed in.
This gets tested here: https://github.com/BigRoomStudios/schwifty/blob/tests/test/index.js#L104-L116

In the test I change `client: 'sqlite3'` to `client: 'fakeConnection'` and it errors as expected. This should probably be tested further to throw nicely if there's an issue connecting to the db, etc.